pretyping/evarconv.ml

```diff
@@ -54,26 +54,15 @@ let debug_ho_unification = CDebug.create ~name:"ho-unification" ()
 (*******************************************)
 (* Functions to deal with impossible cases *)
 (*******************************************)
-let impossible_default_case env =
+let impossible_default_case env sigma =
   let type_of_id = Coqlib.lib_ref "core.IDProp.type" in
-  let c, ctx = UnivGen.fresh_global_instance env (Coqlib.(lib_ref "core.IDProp.idProp")) in
-  let (_, u) = Constr.destRef c in
-  Some (c, Constr.mkRef (type_of_id, u), ctx)
-
-let coq_unit_judge =
-  let open Environ in
-  let make_judge c t = make_judge (EConstr.of_constr c) (EConstr.of_constr t) in
-  let na1 = make_annot (Name (Id.of_string "A")) Sorts.Relevant in
-  let na2 = make_annot (Name (Id.of_string "H")) Sorts.Relevant in
-  fun env ->
-    match impossible_default_case env with
-    | Some (id, type_of_id, ctx) ->
-      make_judge id type_of_id, ctx
-    | None ->
-      (* In case the constants id/ID are not defined *)
```

Comment:
This fallback has been dead code for a while (since about 5cc13770ac2358d583b21f217b8c65d2d5abd850)
The current state of the PR just deletes it, but maybe it's worth resurrecting?

Reply:
I think not, we should not ever fail, but rather report that `core.IDProp.idProp` should be bound.

Reply:
The fallback was intended to be used in `-nois` mode.

Reply:
I guess it used to be harder to make it work in -noinit mode since the constants were looked up by name, now that we use register it should be ok.

Reply:
In the absence of any constants, what can you register?

Reply:
they're easy to define

Reply:
Yes, but that means that this would not work "on the nose" for a user working in a fresh environment (assuming of course that such "purist" users exist).

Reply:
Such purist user can now decide to define it with whatever name he chooses. Same thing for tactics that rely on some constants being defined, not all the system can be used in -nois mode right away, one needs to define a few things. I think that's a fine requirement on the (nonexistent?) purist users.

Reply:
Sure, but it is also pleasant to say that a complex `match` expression does not require any extra ingredients. First, this is somehow a "defect" of the metatheory that "small inversion" is not harwired in the typing rule of a `match`. If it were hardwired, one would not need to fill the impossible cases with dummy terms: the impossible cases would simply not be there in the internal representation. Second, to simulate the impossible cases, `forall A:Prop, A -> A` and `fun (A:Prop) (H:A) => H` are enough. If constants are introduced, it is to shorten these terms, not because constants are needed per se. So, if only one mechanism is kept, I would prefer to get rid of `IDProp` and `idProp` and keep only their expansion.